### PR TITLE
2294 bciers apps e2e

### PR DESCRIPTION
**Pull request by marcellmueller**
#2294

- Adds a `docker-compose-bciers-apps.yaml` file
- Adds `dashboard-e2e` GitHub workflow
- Modify `run-nx-app` action and `test-nx-project-e2e` workflow to accept another docker-compose file
- Modify basic e2e test to take a picture of every app dashboard to show this is working
- Fix reporting dashboard e2e again...I forgot to upsert `CAS_ADMIN` user so the previous fix didn't work. 

#### Local testing
```
cd bc_obps && make run
cd bciers && yarn dev-all
yarn dash:e2e:ui
```

#### Making sure this is working in CI
To verify it's working in CI check out the `cas-dashboard` Happo diffs at the bottom of this PR. 

#### Important:A console.log has been added to trigger nx affected for all apps. It was added for testing/reviewing purposes so we can make sure all e2e jobs are still functioning correctly :)

### CODE REVIEW

File: bciers/libs/components/src/index.ts [hunk 1]
```diff
@@ -6,3 +6,5 @@ export { default as Bread } from "./navigation/Bread";
 export * from "./theme/ThemeRegistry";
 export * from "./theme/EmotionCache";
 export * from "./theme/theme";
+
+console.log("Trigger NX Affected");
```
Comment (marcellmueller) [inline bciers/libs/components/src/index.ts:10]: Note: need to remove this after this has been reviewed :)

File: .github/workflows/main.yaml [hunk 2]
```diff
@@ -68,29 +68,37 @@ jobs:
...
     needs: [build-backend, build-dashboard-e2e, build-coam, check-nx-affected]
     uses: ./.github/workflows/test-nx-project-e2e.yaml
     with:
-      nx_project: coam
-      nx_app_port: 7000
       image_url: "ghcr.io/bcgov/cas-coam-frontend"
       is_nx_affected: ${{contains(needs.check-nx-affected.outputs.NX_AFFECTED_E2E, 'coam') || github.ref_name == 'develop' || github.ref_name == 'main' || github.head_ref == 'develop' || github.head_ref == 'main'}}
+      nx_app_port: 7000
+      nx_project: coam
     secrets: inherit
```
Comment (marcellmueller) [inline .github/workflows/main.yaml:84]: A lot of these changes are me just neurotically putting things in alphabetical order 😅

**Comment by shon-button** [comment]
@marcellmueller 

🎉 

Thanks for doing this, it works great, and certainly can be merged as is but I do have one FYC: 😊

I might have left the `bciers/libs/e2e/src/utils/test-nx-app-landing-page.ts`,  as the simple test for each zone to validate the auth was working for e2e , and hoisted the zone dashboards code into `bciers/apps/dashboard-e2e/tests/dashboards.spec.ts` containing something like:
```
import { test } from "@playwright/test";
// import { analyzeAccessibility } from "@/e2e/utils/helpers";
// ☰ Enums
import { AppRole, UserRole, UserOperatorStatus } from "@/e2e/utils/enums";
// 🥞 DB CRUD
import {
  deleteUserOperatorRecord,
  upsertUserRecord,
  upsertOperatorRecord,
  upsertUserOperatorRecord,
} from "@/e2e/utils/queries";
// ℹ️ Environment variables
import * as dotenv from "dotenv";
dotenv.config({ path: "./e2e/.env.local" });

const happoPlaywright = require("happo-playwright");
test.beforeAll(async () => {
  try {
    // Scenario FrontEndRoles.INDUSTRY_USER_ADMIN where UserOperatorStatus.APPROVED && OperatorStatus.APPROVED;
    // Upsert a User record: bc-cas-dev
    await upsertUserRecord(UserRole.INDUSTRY_USER_ADMIN);
    await upsertUserRecord(UserRole.CAS_ADMIN);
    // Upsert an Operator record, using default values
    await upsertOperatorRecord();
    // Upsert an User Operator record: industry_user_admin, operator id 2
    await upsertUserOperatorRecord(
      process.env.E2E_INDUSTRY_USER_ADMIN_GUID as string,
      AppRole.ADMIN,
      UserOperatorStatus.APPROVED,
    );
    // Scenario FrontEndRoles.INDUSTRY_USER where userOperatorStatus !== UserOperatorStatus.APPROVED
    // Shows "Select Operator\...1 pending action(s) required" bceidSelectOperatorTile
    // ensure user is not associated with any operator
    // Upsert a User record: bc-cas-dev-secondary
    await upsertUserRecord(UserRole.INDUSTRY_USER);
    await deleteUserOperatorRecord(
      process.env.E2E_INDUSTRY_USER_GUID as string,
    );
  } catch (error) {
    // eslint-disable-next-line no-console
    console.error("❌ Error in Db setup for dashboard:", error);
    throw error;
  }
});

test.beforeEach(async ({ context }) => {
  await happoPlaywright.init(context);
});

test.afterEach(async () => {
  await happoPlaywright.finish();
});
// 🏷 Annotate test suite as serial to prevent failure in setupTestEnvironment
test.describe.configure({ mode: "serial" });
const zones = [
  "administration",
  "coam",
  "dashboard",
  "registration",
  "reporting",
];
zones.forEach((zone) => {
  test.describe(`Test BCIERS Dashboard: ${zone}`, () => {
    const url = `${process.env.E2E_BASEURL}${zone}`;
    // Reporting dashboard is broken at the moment
    const user =
      zone === "reporting" ? UserRole.CAS_ADMIN : UserRole.INDUSTRY_USER_ADMIN;
    const testRole = `E2E_${user.toUpperCase()}_STORAGE_STATE`;

    const storageState = JSON.parse(process.env[testRole] as string);

    test.use({ storageState: storageState });

    test("Test Selfie", async ({ page }) => {
      // 🛸 Navigate to landing page
      await page.goto(url);

      // 📷 Cheese!
      const pageContent = page.locator("html");
      await happoPlaywright.screenshot(page, pageContent, {
        component: `Authenticated ${zone} page - ${user}`,
        variant: "default",
      });
    });
  });
});

```

FYI: 
I will update #2311 with some suggested was to reduce the SQL commands and to loop all roles in a file (potentially) called  `bciers/apps/dashboard-e2e/tests/dashboards.spec.ts`

**Review by shon-button — APPROVED** [review]

**Comment by marcellmueller** [comment]
> I might have left the `bciers/libs/e2e/src/utils/test-nx-app-landing-page.ts`, as the simple test for each zone to validate the auth was working for e2e , and hoisted the zone dashboards code into `bciers/apps/dashboard-e2e/tests/dashboards.spec.ts` containing something like:

Thanks, though that is just a placeholder test I created as POC and meant to be deleted/altered as we create actual tests for each app. Not sure it's worth putting more energy into as it's just used to show that this works in CI.

**Pull request merged**